pkg/common/cloudconfig/azure/azure.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// Package azure defines azure cloud provider configuration.
+package azure
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"sigs.k8s.io/cloud-provider-azure/pkg/azclient"
+	"sigs.k8s.io/yaml"
+)
+
+type CloudConfig struct {
+	azclient.ARMClientConfig `json:",inline" mapstructure:",squash"`
+	azclient.AzureAuthConfig `json:",inline" mapstructure:",squash"`
+	// subscription ID
+	SubscriptionID string `json:"subscriptionID,omitempty" mapstructure:"subscriptionID,omitempty"`
+	// azure resource location
+	Location string `json:"location,omitempty" mapstructure:"location,omitempty"`
+	// default resource group where the azure resources are deployed
+	ResourceGroup string `json:"resourceGroup,omitempty" mapstructure:"resourceGroup,omitempty"`
+}
+
+const (
+	// DefaultUserAgent is the default user agent string to access Azure resources.
+	DefaultUserAgent = "fleet-net-controller-manager"
```

Comment:
The default value could be different for hub and member.

It's better to distinguish them